### PR TITLE
tests: Set helm history-max to zero (unlimited) and other fixes for enterprise tests

### PR DESCRIPTION
**Pull request by davidjumani**
# Description

Sets the max history for helm upgrades and rollbacks to 0 (no limit) from the default of 10. This is used by tests that quickly reach the max history when changing and later reverting settings.
This is useful in enterprise as there can be over 30 revisions with the need to revert back to revision 1

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by sam-heilbron — APPROVED** [review]
LGTM.

Would an alternate solution have been to allow defining arbitrary options, so that the tests that needed to defined a custom `history-max` could do so?

**Review by jenshu — DISMISSED** [review]

**Review by jenshu — DISMISSED** [review]

**Comment by davidjumani** [comment]
> Would an alternate solution have been to allow defining arbitrary options, so that the tests that needed to defined a custom history-max could do so?

I don't believe it will be necessary as setting a wrong max history can result in failing to revert to a specific version
Better to be safe and set it to the maximum

File: test/kube2e/util.go [hunk 1]
```diff
@@ -46,13 +46,13 @@ func GetHttpEchoImage() string {
 }
 
 // GlooctlCheckEventuallyHealthy will run up until proved timeoutInterval or until gloo is reported as healthy
-func GlooctlCheckEventuallyHealthy(offset int, testHelper *helper.SoloTestHelper, timeoutInterval string) {
+func GlooctlCheckEventuallyHealthy(offset int, namespace string, timeoutInterval string) {
```
Comment (sam-heilbron) [inline test/kube2e/util.go:49]: I like this change! Better to have the parameters that this function accepts be as minimal as possible.

Given that this is a breaking change to an exported signature, it's worth it to open a PR in enterprise that pulls this in. That way if another developer tries to do the update before you, we have the code/commit that shows what they need to do

**Review by sam-heilbron — APPROVED** [review]

File: changelog/v1.18.0-beta11/set-max-history.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+changelog:
+- type: NON_USER_FACING
+  description: Sets the max history for helm upgrades and rollbacks to 0 (no limit) from the default of 10. This is used by tests that quickly reach the max history when changing and later reverting settings.
```
Comment (nfuden) [inline changelog/v1.18.0-beta11/set-max-history.yaml:3]:
```suggestion
  description: Configures tests to set the max history for helm upgrades and rollbacks to 0 (no limit) from the default of 10. This is used by tests that quickly reach the max history when changing and later reverting settings.
```

File: test/kube2e/util.go (hunk 1 above)
Comment (davidjumani) [reply test/kube2e/util.go:49]: [Enterprise PR](https://github.com/solo-io/solo-projects/pull/6479) with changes incorporated

**Review by sam-heilbron — APPROVED** [review]

**Review by jenshu — APPROVED** [review]

**Pull request merged**